marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/hub-entities.xqy

```diff
@@ -83,6 +83,21 @@ declare function hent:uber-model($models as object-node()*) as map:map
     $uber-model
 };
 
+declare function hent:is-tde-generation-enabled($entity-def as object-node()) as xs:boolean
+{
+  let $entity-def-map as map:map := $entity-def
+  let $definitions := $entity-def-map => map:get("definitions")
+  let $entity-title := $entity-def/info/title
+  (: Check if TDE generation is enabled. Any value other than true enables TDE generation. :)
+  let $primary-type-def := ($definitions => map:get($entity-title), $definitions => map:get(map:keys($definitions)[1]))[1]
+  let $is-property-set := map:contains($primary-type-def, "tdeGenerationDisabled")
+  let $property-val := map:get($primary-type-def, "tdeGenerationDisabled")
+  let $is-property-type-boolean := $is-property-set and string(xdmp:type($property-val)) = "boolean"
```

Comment:
Hmm - if the code is simpler in XQuery to just support "true" and true, let's do that (I don't think I've ever used xdmp:type before, so this seems more complex than it should be).